ResidentInformationApi.Tests/V1/Gateways/AcademyInformationGatewayTests.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using FluentAssertions;
+using Moq;
+using Moq.Protected;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Boundary.Responses;
+using ResidentInformationApi.V1.Gateways;
+
+namespace ResidentInformationApi.Tests.V1.Gateways
+{
+    [TestFixture]
+    public class AcademyInformationGatewayTests
+    {
+        private Fixture _fixture;
+        private AcademyInformationGateway _classUnderTest;
+        private Mock<HttpMessageHandler> _messageHandler;
+        private Uri _uri;
+        private String _currentEnv;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _fixture = new Fixture();
+            _uri = new Uri("http://test-domain-name.com/");
+            _currentEnv = Environment.GetEnvironmentVariable("ACADEMY_API_ENDPOINT");
+            Environment.SetEnvironmentVariable("ACADEMY_API_ENDPOINT", _uri.OriginalString);
+            _messageHandler = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+
+            var httpClient = new HttpClient(_messageHandler.Object)
+            {
+                BaseAddress = _uri,
+            };
+
+            _classUnderTest = new AcademyInformationGateway(httpClient);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Environment.SetEnvironmentVariable("ACADEMY_API_ENDPOINT", _currentEnv);
+        }
+
+        [Test]
+        public async Task GetClaimantInformationReturnsEmptyArrayIfNoResultsFound()
+        {
+            var rqp = new ResidentQueryParam();
+            SetUpMessageHandlerToReturnJson("[]");
+            var received = await _classUnderTest.GetClaimantInformation(rqp);
+
+            received.Should().BeNullOrEmpty();
```

Comment:
Maybe we should be specific here about whether the response is null or empty

Reply:
Updated this test to show that the returned value should be empty but should not be null